### ISSUE TITLE
source files with unmatched build constraints will be ignored by vim-go and cause errors

### ISSUE THREAD

**Issue by steveeJ** [fatih/vim-go#822]
### Actual behavior

After adding a custom tag as a build constraint in a `.go` file, which could look like this

```
// +build myCustomTag
```

vim-go no longer functions properly. When trying to jump to a definition an error occurs:

```
vim-go: guru: no buildable Go source files in /home/steveej/src/github/coreos/rkt/vimgo
```

In this case I created a package in an existing project directory with only this new file with the build constraint that is not matched.

### Expected behavior

vim-go should be able to work with source code despite the build constraints.

### Steps to reproduce:

Please create a reproducible case of your problem. Re produce it 
with a minimal `vimrc` with all plugins disabled and only `vim-go`
enabled:

1. Create a minimal go package. This would work:

    ```go
// +build aTag

package aPackage

import "fmt"
                                                                                                                             
func main() { fmt.Println("hello vim-go") } 
```

2. Open vim with vim-go enabled
3. try to :GoDef fmt.Println

### Configuration

* vim version: NVIM 0.1.2
* vim-go version: rev b3fddb60fdab980d1fc339fbcbc879abd6cbf524
* go version: go version go1.6.2 linux/amd64

**Comment by sillyousu** [comment]
You can set tags manually:
`let g:go_guru_tags="aTag" in vim vimrc or  `:GoGuruTags aTag`

And then apply this patch:
https://github.com/fatih/vim-go/pull/821

**Comment by fatih** [comment]
@steveeJ that's normal. `guru` uses the loader to load a valid go program, and if you don't tell them it'll not work. The PR #821 allows you to pass the necessary tags either statically via a variable or dynamically via the `:GoGuruTags` command

**Comment by fatih** [comment]
Fixed with #821
